File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/VB/DevExpress.SalesDemo.Wpf/ViewModel/SalesDashboardViewModel.vb
Imports DevExpress.Mvvm.POCO

Namespace DevExpress.SalesDemo.Wpf.ViewModel

    Public Class SalesDashboardViewModel
        Inherits PageViewModel

        Public Shared Function Create() As SalesDashboardViewModel
            Return ViewModelSource.Create(Function() New SalesDashboardViewModel())
        End Function

        Protected Sub New()
        End Sub

        Public Overridable Property SalesSummaryViewModel As SalesSummaryViewModel

        Public Overridable Property DailySalesPerformanceViewModel As PerformanceAreaChartViewModel

        Public Overridable Property MonthlySalesPerformanceViewModel As PerformanceAreaChartViewModel

        Protected Overrides Sub Initialize()
            SalesSummaryViewModel = SalesSummaryViewModel.Create()
            DailySalesPerformanceViewModel = PerformanceAreaChartViewModel.Create(Dashboard, PerformanceViewMode.Daily)
            MonthlySalesPerformanceViewModel = PerformanceAreaChartViewModel.Create(Dashboard, PerformanceViewMode.Monthly)
        End Sub
    End Class
End Namespace

File: DevExpress-Demo/DevExpress.SalesDemo.Wpf/CS/DevExpress.SalesDemo.Wpf/ViewModel/SalesDashboardViewModel.cs
using DevExpress.Mvvm.POCO;

namespace DevExpress.SalesDemo.Wpf.ViewModel {
    public class SalesDashboardViewModel : PageViewModel {
        public static SalesDashboardViewModel Create() {
            return ViewModelSource.Create(() => new SalesDashboardViewModel());
        }
        protected SalesDashboardViewModel() { }

        public virtual SalesSummaryViewModel SalesSummaryViewModel { get; protected set; }
        public virtual PerformanceAreaChartViewModel DailySalesPerformanceViewModel { get; protected set; }
        public virtual PerformanceAreaChartViewModel MonthlySalesPerformanceViewModel { get; protected set; }
        protected override void Initialize() {
            SalesSummaryViewModel = SalesSummaryViewModel.Create();
            DailySalesPerformanceViewModel = PerformanceAreaChartViewModel.Create(Modules.Dashboard, PerformanceViewMode.Daily);
            MonthlySalesPerformanceViewModel = PerformanceAreaChartViewModel.Create(Modules.Dashboard, PerformanceViewMode.Monthly);
        }
    }
}
